apps/clockcal/app.js

```diff
@@ -211,8 +212,11 @@ function action(a) {
     g.reset();
     if (typeof secondInterval !== "undefined") clearTimeout(secondInterval);
     if (DEBUG) console.log("action:" + a);
+    state = "unknown";
+    console.log("state -> unknown");
     switch (a) {
         case "[ignore]":
+            drawWatch();
```

Comment:
Is a `drawWatch()` needed here? I'm not against it, but seems unnecessary too me (I might overlook something or have a misconception).

Reply:
kinda but not really: I was playing with different states (so the hardware button would do different things: when I am showing the calendar it should redraw the watch, if I am in the watch, it should open the menu.
So I change the state to "unknown" when the launcher is called and whenever I draw the watch, I'm back in the watch-state. Somehow the app hangs when I try to go to drawWatch() after an showAlert(), but I wanted to investigate that further in the future. That's why I left the state = unknown in and call the drawWatch() in the [ignore].
Still things to do, but for now I think it doesn't hurt too much.